File: Python/algv2.py
#!/usr/bin/python3

import sys
import re
from datetime import datetime

def parseARGV(args):
  if args is None:
    return None
  arg=args[1]
  v=[]
  m=re.match(r'^-?\d+$',arg)
  if m is not None:
    n=len(args)
    for i in range(1,n):
      v.append(int(args[i]))
    return v
  infile=sys.stdin
  m=re.match(r'^-?-?[Ss][Tt][Dd](?:[Ii][Nn])?$',arg)
  cap=0
  if m is None:
    infile=open(arg,'r',encoding="utf-8")
  if len(sys.argv)>2:
    arg=args[2]
    m=re.match(r'^[1-9]\d*$',arg)
    if m is not None:
      cap=int(arg)
      v=[0]*cap
  if cap==0:
    for line in infile:
      v.append(int(line))
    return v
  x=0
  for line in infile:
    v[x]=int(line)
    x=x+1
    if x==cap:
      break
  for line in infile:
    v.append(int(line))
    x=x+1
  if x<cap:
    del v[x:]
  return v

#https://www.youtube.com/watch?v=Qgevy75co8c

def somaMaximaV2(v):
  if len(v)==0:
    return None

  n=len(v)
  #solucao do caso de somente um elemento
  r=v[0]
  #para as demais as posicoes iniciais
  for i in range(1,n):
    s=0
    #para todas as posicoes finais de forma descrescente
    for j in range(i,-1,-1):
      #realiza a soma parcial e compara com o melhor ate o momento
      s=s+v[j]
      if s>r:
        r=s
  return r

if len(sys.argv)<2:
  print(f'{sys.argv[0]} : vetor\n')
  sys.exit(1)

preparse=datetime.now()
v=parseARGV(sys.argv)
if v is None:
  print(f'{sys.argv[0]} : ERRO no parse de ({sys.argv[1]})\n')
  sys.exit(2)

precalc=datetime.now()
somaMaxima=somaMaximaV2(v)
poscalc=datetime.now()

aloctime=precalc-preparse
algtime=poscalc-precalc

print(f'\n{sys.argv[0]} : Tempo Alocacao memoria vetor ({aloctime.total_seconds()})')
print(f'\n{sys.argv[0]} : Tempo Calculo da soma maxima ({algtime.total_seconds()})')
print(f'\n{sys.argv[0]} : A soma maxima do array com ({len(v)}) posicoes eh ({somaMaxima})')

File: Java/algv2.java
import java.lang.*;
import java.util.*;
import java.io.*;
import java.io.File;
import java.util.regex.Pattern;
import java.time.*;

public class algv2
{
	public static int len=0;
	public static int cap=0;

	public static long[] parseARGV(String[] args)
	{
		String arg=args[0];
		long[] v=null;
		int n=args.length;
		boolean m=Pattern.matches("^-?\\d+$",arg);
		if(m)
		{
			int i;
			v=new long[n];
			for(i=0;i<n;++i)
			{
				try
				{
					long x=Long.parseLong(args[i]);
					v[i]=x;
				}
				catch(NumberFormatException e)
				{
					return null;
				}
			}
			len=n;
			cap=n;
			return v;
		}
		m=Pattern.matches("^-?-?[Ss][Tt][Dd](?:[Ii][Nn])?$",arg);
		Scanner s=null;
		if(m)
		{
			s=new Scanner(System.in);
		}
		else
		{
			try
			{
				s=new Scanner(new File(arg));
			}
			catch(FileNotFoundException e)
			{
				return null;
			}
		}
		int cp=16;
		if(n>1)
		{
			arg=args[1];
			m=Pattern.matches("^[1-9]\\d*$",arg);
			if(m)
			{
				try
				{
					int x=Integer.parseInt(arg);
					cp=x;
				}
				catch(NumberFormatException e)
				{
					s.close();
					return null;
				}
			}
		}
		s.useDelimiter("\n");
		int size=0;
		v=new long[cp];
    while(s.hasNextLong())
		{
			long x=s.nextLong();
			if(size==cp)
			{
				cp<<=1;
				long[] newv=Arrays.copyOf(v,cp);
				v=newv;
			}
			v[size++]=x;
    }
	s.close();
		len=size;
		cap=cp;
		return v;
	}

	public static long somaMaximaV2(long[] v)
	{
		int n=len;
	  if(n==0)
		{
	    return 0L;
		}

		long s=0L;
		//solucao do caso de somente um elemento
	  long r=v[0];
	  //para as demais as posicoes iniciais
	  for(int i=0;i<n;++i)
		{
			s=0;
	    //para todas as posicoes finais de forma descrescente
	    for(int j=i;j>=0;--j)
			{
	      //realiza a soma parcial e compara com o melhor ate o momento
	      s=s+v[j];
	      if(s>r)
				{
	        r=s;
				}
			}
		}
	  return r;
	}

	public static void main(String args[])
	{
		//String execfile=new java.io.File(algv1.class.getProtectionDomain().getCodeSource().getLocation().getPath()).getName();
		String execfile=algv2.class.getName();

		if(args.length<1)
		{
			System.out.println("\n"+execfile+" : vetor\n");
			System.exit(1);
		}

		Instant preparse=Instant.now();
		long[] v=parseARGV(args);
		if(v==null)
		{
			System.out.println("\n"+execfile+" : ERRO no parse de ("+args[0]+")\n");
			System.exit(2);
		}

		Instant precalc=Instant.now();
		//System.out.println("\n"+execfile+" : V ("+Arrays.toString(v)+")\n");
		long somaMaxima=somaMaximaV2(v);
		Instant poscalc=Instant.now();

		Duration parse=Duration.between(preparse,precalc);
		Duration alg=Duration.between(precalc,poscalc);

		System.out.printf("\n%s : Tempo Alocacao memoria vetor (%d.%09d)\n",execfile,parse.getSeconds(),parse.getNano());
		System.out.printf("\n%s : Tempo Calculo da soma maxima (%d.%09d)\n",execfile,alg.getSeconds(),alg.getNano());
		System.out.printf("\n%s : A soma maxima do array com (%d) posicoes eh (%d)\n",execfile,len,somaMaxima);
		System.exit(0);
	}
}
